heat/core/statistics.py

```diff
@@ -1103,6 +1135,236 @@ def mpi_argmin(a, b, _):
 MPI_ARGMIN = MPI.Op.Create(mpi_argmin, commute=True)
 
 
+def percentile(x, q, axis=None, interpolation="linear", keepdim=False):
+    """
+    Compute the q-th percentile of the data along the specified axis.
+    Returns the q-th percentile(s) of the tensor elements.
+
+    Parameters:
+    -----------
+
+    x : DNDarray
+        Input tensor
+
+    q : DNDarray, scalar, or list of scalars
+        Percentile or sequence of percentiles to compute. Must belong to the interval [0, 100].
+
+    axis : int, or None, optional #TODO tuple of ints
+        Axis along which the percentiles are computed. Default is None.
+
+    #TODO: out, optional. Output buffer.
+
+    interpolation : {‘linear’, ‘lower’, ‘higher’, ‘midpoint’, ‘nearest’}, optional
+        Interpolation method to use when the desired percentile lies between two data points i < j:
+
+        ‘linear’: i + (j - i) * fraction, where fraction is the fractional part of the index surrounded by i and j.
+        ‘lower’: i.
+        ‘higher’: j.
+        ‘nearest’: i or j, whichever is nearest.
+        ‘midpoint’: (i + j) / 2.
+
+    keepdim : bool, optional
+        If True, the axes which are reduced are left in the result as dimensions with size one.
+        With this option, the result can broadcast correctly against the original array x.
+
+    Returns:
+    --------
+
+    percentile : process-local DNDarray
+
+    """
+
+    def local_percentile(data, axis, indices):
+        """
+        Process-local percentile calculation.
+
+        Input:
+        ------
+        data : torch.tensor
+        axis : int or TODO tuple of ints
+        indices : torch.tensor
+
+        Returns:
+        --------
+        percentile, torch.tensor, process-local
+        """
+
+        axis_slice = data.ndim * (slice(None, None, None),)
+        if indices.dtype == torch.long or indices.dtype == torch.int:
```

Comment:
Better check here for identity (faster) with the 'is' operator

Reply:
Done!